### ISSUE TITLE
Regent: max/min component-wise for int3d

### ISSUE THREAD

**Issue by syamajala** [StanfordLegion/legion#784]
Doing the following:
```
import "regent"
local c = regentlib.c

task main()
    var a = int3d{2, 8, 10}
    var b = int3d{1, 4, 11}
    var d = max(a, b)
    c.printf("%d %d %d\n", d.x, d.y, d.z)
    d = min(a, b)
    c.printf("%d %d %d\n", d.x, d.y, d.z)
end

regentlib.start(main)
```
results in b, but I was expecting int3d{2, 8, 11}. Similarly, `min(a, b)` results in b instead of int3d{1, 4, 10}.

**Comment by streichler** [comment]
Without knowing the types of `a` and `b`, if you told me that `min(a,b)` or `max(a,b)` can return something other than `a` or `b`, I'd be pretty surprised...

So if we were going to offer a per-component-min/max (instead of just telling the programmer to roll their own), I'd want it to have a different name.

**Comment by syamajala** [comment]
We could do what numpy does and call the element-wise operations maximum and minimum. The problem is we don't have any documentation on the regent standard library so there would be no easy way for anyone to find them or even know they exist without asking someone.

**Comment by elliottslaughter** [comment]
Thinking about this more, I think @streichler's view makes sense. Otherwise `min` wouldn't be consistent with `<`, etc.

You can define your own minimum/maximum with inline tasks:

```
__demand(__inline)
task minimum(a : int3d, b : int3d)
  return int3d { min(a.x, b.x), min(a.y, b.y), min(a.z, b.z) }
end
```

It's not super pretty but it should have the desired effect and shouldn't have an impact on performance (since it'll be inlined).

**Comment by streichler** [comment]
@elliottslaughter How would you feel about creating and curating a "regent standard library" as part of the main repo with things like this and maybe the file I/O and temporary-array-construction we talked about yesterday?

**Comment by elliottslaughter** [comment]
Yeah, I think this is a good idea. I'd like @syamajala's BLAS bindings to be there too.

One issue with "minimum" as it's currently written is that it can't be polymorphic. So we'd need to either add such support to Regent or else make a version for each dimension. One advantage of having that particular function be in the user's program is that the user will presumably know exactly what dimension(s) they need.

**Comment by syamajala** [comment]
I have some other ideas for things that would be nice to have in a standard library. I'll open another ticket so we can have a discussion there instead of having it get lost here.
